projects/generate_glitches/generate_glitches.py

```diff
@@ -0,0 +1,342 @@
+
+    Arguments:
+    - ifo: ifo to generate glitch triggers for
+    - snr_thresh: snr threshold above which to keep as glitch
+    - start: start gpstime
+    - stop: stop gpstime
+    - window: half window around trigger time to query data for
+    - sample_rate: sampling frequency
+    - channel: channel name used to read data
+    - frame_type: frame type for data discovery w/ gwdatafind
+    - trig_file: txt file output from omicron triggers
+            (first column is gps times, 3rd column is snrs)
+    - vetoes: SegmentList object of times to ignore
+    """
+    print(gwdatafind.find_urls)
+    glitches = []
+    snrs = []
+
+    # snr and time columns in omicron file
+    snr_col = 2
+    time_col = 0
+
+    # load in triggers
+    triggers = np.loadtxt(trig_file)
+
+    # if passed, apply vetos
+    if vetoes is not None:
+        keep_bools = veto(triggers[:, time_col], vetoes)
+        triggers = triggers[keep_bools]
+
+    # restrict triggers to within gps start and stop times
+    times = triggers[:, time_col]
+    time_args = np.logical_and(times > start, times < stop)
+    triggers = triggers[time_args]
+
+    # apply snr thresh
+    day_snrs = triggers[:, snr_col]  # second column is snrs
+    snr_thresh_args = np.where(day_snrs > snr_thresh)
+    triggers = triggers[snr_thresh_args]
+
+    # re-set 'start' and 'stop' so we aren't querying unnecessary data
+    start = np.min(triggers[:, time_col]) - 2 * window
+    stop = np.max(triggers[:, time_col]) + 2 * window
+
+    logging.info(
+        f"Querying {stop - start} seconds of data for {len(triggers)} triggers"
+    )
+
+    # use gwdatafind to create frame cache
+    frames = gwdatafind.find_urls(
+        site=ifo.strip("1"),
+        frametype=f"{ifo}_{frame_type}",
+        gpsstart=int(start),
+        gpsend=int(stop),
+        urltype="file",
+        on_gaps="ignore",
+    )
+
+    # read frames all at once, then crop
+    # this time series for each trigger.
+    # Although its initially slower to read all the frames at once,
+    # I think this is overall faster than querying a 4096 frame for
+    # every trigger.
+
+    ts = TimeSeries.read(
+        frames, channel=f"{ifo}:{channel}", start=start, end=stop, pad=0
+    )
+
+    # for each trigger
+    for trigger in tqdm(triggers):
+        time = trigger[time_col]
+
+        try:
+            glitch_ts = ts.crop(time - window, time + window)
+
+            glitch_ts = glitch_ts.resample(sample_rate)
+
+            snrs.append(trigger[snr_col])
+            glitches.append(glitch_ts)
+
+        except ValueError:
+            logging.info(f"Data not available for trigger at time: {time}")
+            continue
+
+    glitches = np.array(glitches)
+    snrs = np.array(snrs)
+    return glitches, snrs
+
+
+@typeo
+def main(
+    snr_thresh: float,
+    start: float,
+    stop: float,
+    window: float,
+    omicron_dir: str,
+    out_dir: str,
```

Comment:
Minor point but you might consider using `pathlib.Path` annotations for these. `typeo` will pick that up and map them to `Path`s before passing them in, and then you can make use of all the nice `pathlib` syntax like using the `/` operator (not string) for separating path components and even the `Path.glob` method.